back/engines/commercial/content_builder/spec/acceptance/homepage_layouts_spec.rb

```diff
@@ -0,0 +1,165 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'rspec_api_documentation/dsl'
+
+resource 'ContentBuilderLayouts' do
+  explanation 'Content builder layouts for projects.'
+
+  before { header 'Content-Type', 'application/json' }
+
+  context 'when not signed in' do
+    let!(:homepage) { create(:home_page) }
+    let(:code) { 'homepage' }
+    let(:layout) { homepage.content_builder_layouts.find_by(code: code) }
+
+    get 'web_api/v1/home_pages/content_builder_layouts/:code' do
+      example_request 'Get one layout by code' do
+        assert_status 200
+
+        json_response = json_parse(response_body)
+        expect(json_response).to include(
+          {
+            data: {
+              id: layout.id,
+              type: 'content_builder_layout',
+              attributes: hash_including(
+                code: code,
+                created_at: match(time_regex),
+                updated_at: match(time_regex)
+              )
+            }
+          }
+        )
+      end
+    end
+
+    post 'web_api/v1/home_pages/content_builder_layouts/:code/upsert' do
+      example_request '[error] Try to update a layout without authorization' do
+        assert_status 401
+      end
+    end
+
+    delete 'web_api/v1/home_pages/content_builder_layouts/:code' do
+      example_request '[error] Try to delete a layout without authorization' do
+        assert_status 401
+      end
+    end
+  end
+
+  context 'when admin' do
+    before { admin_header_token }
+
+    let!(:homepage) { create(:home_page) }
+    let(:code) { 'homepage' }
+
+    context 'when the layout exists' do
+      let(:layout) { homepage.content_builder_layouts.find_by(code: code) }
+
+      get 'web_api/v1/home_pages/content_builder_layouts/:code' do
+        example_request 'Get one layout by code' do
+          assert_status 200
+        end
+      end
+
+      get 'web_api/v1/home_pages/content_builder_layouts/unknown' do
+        example_request '[error] Try to get a layout when the code is unknown' do
+          assert_status 404
+        end
+      end
+
+      post 'web_api/v1/home_pages/content_builder_layouts/:code/upsert' do
```

Comment:
We've used a `PUT` in other places where we're doing an upsert - I think it would be the more technically correct HTTP method. Probably wouldn't need the upsert part of the path then too

Reply:
Makes sense to me. But I kept the original implementation. I would say there is some tech debt around the content builder layouts:
- Using one API for all layouts (currently report layouts have their own API)
- Removing the engine.